assets/app/scripts/controllers/events.js

```diff
@@ -8,22 +8,76 @@
  * Controller of the openshiftConsole
  */
 angular.module('openshiftConsole')
-  .controller('EventsController', function ($routeParams, $scope, DataService, ProjectsService, Logger) {
+  .controller('EventsController', function ($filter, $routeParams, $scope, DataService, ProjectsService, Logger) {
     $scope.projectName = $routeParams.project;
-    $scope.emptyMessage = "Loading...";
-    $scope.events = {};
+    $scope.filter = {
+      text: ''
+    };
     $scope.renderOptions = {
       hideFilterWidget: true
     };
-    var watches = [];
+
+    $scope.unfilteredEvents = {};
+
+    var sortedEvents = [];
+    var sortEvents = function() {
+      // TODO: currentField is renamed in angular-patternfly 3.0
+      var sortID = _.get($scope, 'sortConfig.currentField.id', 'lastTimestamp'),
+          order = $scope.sortConfig.isAscending ? 'asc' : 'desc';
+      sortedEvents = _.sortByOrder($scope.unfilteredEvents, [sortID], [order]);
+    };
+
+    var applyFilter = $filter('filter');
+    var filterEvents = function() {
+      $scope.filteredEvents = applyFilter(sortedEvents, $scope.filter.text);
+    };
+    $scope.$watch('filter.text', filterEvents);
+
+    var update = function() {
+      // Sort first so we can update the filter as the user types without resorting.
+      sortEvents();
+      filterEvents();
+    };
+
+    $scope.sortConfig = {
+      fields: [{
+        id: 'lastTimestamp',
+        title: 'Time',
+        sortType: 'alpha'
+      }, {
+        id: 'involvedObject.name',
+        title: 'Name',
+        sortType: 'alpha'
+      }, {
+        id: 'involvedObject.kind',
+        title: 'Kind',
+        sortType: 'alpha'
+      }, {
+        id: 'type',
+        title: 'Severity',
+        sortType: 'alpha'
+      }, {
+        id: 'reason',
+        title: 'Reason',
+        sortType: 'alpha'
+      }, {
+        id: 'message',
+        title: 'Message',
+        sortType: 'alpha'
+      }],
+      isAscending: false,
+      onSortChange: update
+    };
 
     ProjectsService
       .get($routeParams.project)
       .then(_.spread(function(project, context) {
         $scope.project = project;
-        watches.push(DataService.watch("events", context, function(events) {
-          $scope.events = events.by("metadata.name");
-          $scope.emptyMessage = "No events to show";
+
+        var watches = [];
+        watches.push(DataService.watch("events", context, function(data) {
+          $scope.unfilteredEvents = data.by("metadata.name");
+          update();
```

Comment:
Should we debounce the call to update?  Just worried about how often we are going to be re-sorting and re-filtering this list in cases where we are getting spammed with events, especially repeating events where the count just keeps increasing